<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>RNA-Seq and Transcriptomics </title>
	<link rel="stylesheet" type="text/css" href="css/styles.css">


</head>
<body>

<div class="top-bar" >

	<h1 align="center">INTRODUCTION TO TUBERCULOSIS GENOMIC EPIDEMIOLOGY</h1>
</div>

<div class="container1" >
	<p>

	</p>
<h1  align="center">::RNA-Seq and Transcriptomics::</h1>
<h3  align="center">6.1 Case-Study: <i>Klebsiella pneumoniae</i> Transcriptomics - Relationship between the two-component response regulator OmpR and hypermucoviscosity</h3>
</div>
<div class="container1" >
<h1  align="right">
	<p>

	</p>

</h1>
</div>

<a href="index.html" target="_blank"><<-HOME</a>
<div>
<object data="pdfs/M6_RNAseq.pdf#page=1" type="application/pdf" width="100%" height="500px">
          <p><b>Example fallback content</b>: This browser does not support PDFs. Please download the PDF to view it:
          <a href="pdfs/M6_RNAseq.pdf">Download PDF</a>.</p>
</object>
</div>


<div class="container2" >
<h1  align="right">
	<p>

	</p>
	<p>

	</p>
</h1>
</div>
<p>  </p>
<p>  </p>
<p>  </p>
<p> Click <a href="pdfs/Course_doc_all.pdf.pdf" target="_blank">here</a> to download the full course documentation conatining this tutorial.</p>

<br>
<h2>Introduction</h2>

<p>Hypervirulent <i>Klebsiella pneumoniae</i> (hvKp) entails a distinct pathotype that contrasts with classical Klebsiella pneumoniae (cKp) in its ability and proclivity to cause invasive disease in otherwise healthy individuals [1, 2]. hvKp are often described at multiple sites owing to its metastatic spread and associated with infection acquired in the community. Liver abscess is considered one of its hallmark clinical traits [3].
One of the phenotypic traits associated with hvKp is the hypermucoviscosity phenotype, usually determined by a positive string test upon growth in blood agar [4]. Different genetic determinants have been associated with this specific phenotype, mainly the ability to produce the regulator of mucoid phenotype RmpA [5].  This practical session covers a partial analysis of the transcriptomic data generated by Wang et al (2023) which further elucidates the role of OmpR regulator in the hypermucoviscosity phenotype [6].</p>

<p>Citation:<p>
<p>Wang L, Huang X, Jin Q, Tang J et al. Two-Component Response Regulator OmpR Regulates Mucoviscosity through Energy Metabolism in <i>Klebsiella pneumoniae</i>. Microbiol Spectr 2023 Jun 15;11(3):e0054423. PMID: 37097167  http://doi: 10.1128/spectrum.00544-23</p>


<p>The data for this practical is available through the NCBI Gene Expression Omnibus (GEO) database under accession GSE197039. The dataset comprises four samples, subjected to RNA-Seq by paired-end sequencing, representing a ompR knockout mutant and the wild-type strain (two biological replicates each, Table 1):</p>

<img src="images/kp_table1.png" alt="blank space", style="max-width: 70%;">

<p>The practical is divided in three exercises, Exercise 1 pertains the mapping and generation of raw transcript count and has been carried out ahead since the computational steps involved are time consuming. Nonetheless, the commands executed are outlined below along with explanatory context. The second exercise will involve the visualization of mapped reads and the third exercise will cover the identification of differentially expressed genes across the conditions tested.</p>
<br>
<br>
<h2>Exercise 1 – Mapping and Raw read Count</h2>

<p>The initial stage of the analytical process of RNA-Seq data for this practical will consist of mapping the reads to a reference genome following a similar approach as the one already covered in the mapping module for this course. As such, this will require an adequate genome reference file. An annotation file covering all genes in this same reference genome will also be needed. We will use the genome of <i>K. pneumoniae</i> as reference genome, which has already been downloaded in the FASTA format, GenBank format and annotation (GFF3) from NCBI GenBank (accession CP009208). The following files are therefore already present in the course module directory in the Kp subdirectory. Let’s start!</p>

<p>From the home directory type:</p>

<pre>
$ cd Module6/Kp

$ ls
</pre>

<p>Please note for the following files:</p>
<p>•	CP009208_1.fasta – reference genome in the fasta format;</p>
<p>•	CP009208_1.gb  - reference genome in the GenBank format;</p>
<p>•	CP009208_1.gff3 – annotation file in the GFF3 format.</p>
<p>You can look inside these files to confirm its format.</p>



<p>In the same directory you will find compressed FastQ files containing the RNA-Seq raw reads. Each is named according to the Run ID in Table 1 and if this is paired-end sequencing, it will have _1.fastq.gz or _2.fastq.gz appended to the Run ID.</p>
<p>Which one do you have? ____________________________</p>

<p>The mapping steps were done using the following commands:</p>



<pre>
$ bwa mem CP009208_1.fasta SRR18075271_1.fastq.gz SRR18075271_2.fastq.gz | samtools sort --write-index -o SRR18075271.bam -
$ bwa mem CP009208_1.fasta SRR18075272_1.fastq.gz SRR18075272_2.fastq.gz | samtools sort --write-index -o SRR18075272.bam -
$ bwa mem CP009208_1.fasta SRR18075273_1.fastq.gz SRR18075273_2.fastq.gz | samtools sort --write-index -o SRR18075273.bam -
$ bwa mem CP009208_1.fasta SRR18075273_1.fastq.gz SRR18075273_2.fastq.gz | samtools sort --write-index -o SRR18075273.bam –
</pre>

<p>Each command allows for mapping of the reads using BWA MEM algorithm and pipes it to samtools in order generate a BAM containing all mapped reads and reference. Next you need to index the BAM file by typing:</p>
<pre>
$ samtools index SRR18075271.bam
$ samtools index SRR18075272.bam
$ samtools index SRR18075273.bam
$ samtools index SRR18075274.bam
</pre>

<p>At this point you can already visualize your mapping data but we will leave it to the second part of this practical. For now we need to count the number of reads mapped to each feature/gene. Please note that for this stage you will need to know the position, i.e., start, stop and strand of the genes in your gene. This information is not present in the FASTA reference genome you used to map the reads nor you can extract it directly from the BAM files. This is where the GFF file is needed! You will be using HTSeq to count the reads in the genes, specifically its htseq-count script [7]. By typing:</p>


<pre>
$ htseq-count
#or
$ htseq-count --help
</pre>

<p>You will learn a little bit of the different files needed, arguments to this command and output. You can produce the files containing the raw counts by doing the following:</p>
<pre>
$ htseq-count -f bam -r pos -s reverse -t gene -i Name SRR18075271.bam CP009208_1.gff3 > SRR18075271_count.txt
$ htseq-count -f bam -r pos -s reverse -t gene -i Name SRR18075272.bam CP009208_1.gff3 > SRR18075272_count.txt
$ htseq-count -f bam -r pos -s reverse -t gene -i Name SRR18075273.bam CP009208_1.gff3 > SRR18075273_count.txt
$ htseq-count -f bam -r pos -s reverse -t gene -i Name SRR18075274.bam CP009208_1.gff3 > SRR18075274_count.txt
</pre>

<p>Can you check why are we using those extra arguments besides the BAM and the GFF file?
Also, please note that the output is being redirected to files named *_count.txt files. Look at their content. Does it resemble something like this:</p>

<img src="images/counts.png" alt="blank space", style="max-width: 70%;">

<p>These files contain the raw read counts for each gene and will be used as input for the Differential Gene Expression analysis in Exercise 3. But before that let us visualize the mapped data so that you can understand better what we have done so far and what these counts represent.</p>

<p>Notice that you already have the files with the raw counts! To download all previously created BAM files and index files for this practical session, please go to: </p>

<a href="https://ulisboa-my.sharepoint.com/:f:/g/personal/jperdigao_office365_ulisboa_pt/EkBjH_nIesdIlqBXPIXy3_oB4NG7dg4RAksXHik2lODmbQ?e=fyu1w7">https://ulisboa-my.sharepoint.com/:f:/g/personal/jperdigao_office365_ulisboa_pt/EkBjH_nIesdIlqBXPIXy3_oB4NG7dg4RAksXHik2lODmbQ?e=fyu1w7</a> 

<p></p>
<p>and download the bam_files_KP.tar.gz file (use command: tar -xvf bam_files_KP.tar.gz to decompress the file). Run this last command in the Kp sub-directory after downloading the file to this location.</p>
<br>
<br>
<h2>Exercise 2 – Visualization of RNA-Seq mapped data</h2>
<p>For the next exercise you can visualize the mapped reads in Artemis. Inside the directory containing the data, type:</p> 
<pre>
$ art
</pre>
<p>This will start Artemis. You must first load your reference genome, which can be done in multiple ways:</p>

<p>i) you can open the CP009208_1.gb file, containing the genome and features in the GenBank file, or;</p>
<p>ii) you can open the  CP009208_1.fasta file and then read in the GFF file so that you can see the different feature tracks.</p>


<p>Next, you need to read in your BAM files. You can start by going to File >> Read BAM/VCF and select the first BAM file from the list. Please remember that to load a BAM file in Artemis you need to have the BAM file and the respective .bai index file in the same directory. If everything went ok a BAM window with the mapped reads should appear on top of the genome:</p>

<img src="images/bam_viewKp.png" alt="blank space", style="max-width: 70%;">

<p>Depending on whether the genome map window is too zoomed in or out you will see the individual reads or a plot. You can change the zoom level in the vertical scroll bar on the right and by right clicking on the BAM window you can also select the type of graph under View (try changing between pileup and coverage for example).</p>

<img src="images/bam_viewKp2.png" alt="blank space", style="max-width: 70%;">

<p>Let’s look at a particular gene, for example <i>mhpA</i>, by using the Navigator (Goto >> Navigator) and typing the gene name in the appropriate search box.</p>

<img src="images/mphA.png" alt="blank space", style="max-width: 70%;">

<p>Notice that in the vicinity of these genes, there are some with no/low coverage while others have coverage and there are even reads mapping to intergenic regions. Why is that?</p>

<p>Next, you can add and compare the coverage from other BAM files. To do this just right click on the BAM window, select “Add BAM” and add the different BAM files. Also, changing the Graph type to coverage will facilitate the interpretation of the Data. By right clicking on the BAM window you also have the option to configure these lines (changing the colour, thickness, etc.).</p>

<img src="images/atp_operon.png" alt="blank space", style="max-width: 70%;">

<p>Can you find any differentially expressed genes (DEG) using this approach?</p>
<p>Hint: Try the <i>atp</i> operon region!</p>
<br>
<br>

<h2>Exercise 3 – Examining Differential Expression</h2>

<p>This last exercise is aimed at identifying differentially expressed genes (DEGs) by taking the raw counts as input to software that normalizes the number of mapped reads and apply statistical tests to identify DEGs. The most widely used packages for this purpose are the R packages DESeq2 and EdgeR. In this practical we will use DESeq2 to identify DEGs, but code DEG identification using EdgeR is also provided and you can even compare the results between packages [8, 9].</p>
<p>It is also important to stress that using different replicates is a fundamental aspect of DEG analysis as expression levels are variable and depend on several external factors. Biological replicates are therefore essential to account for the variability of biological systems and more accurately compare expression levels between groups/conditions.</p>

<p>The exercise will be carried out in R which involves a different programming language. Let’s go step by step.
Let’s start R by typing:</p>
<pre>
$ R
</pre>
<p>Inside the R command line we need to load the required libraries for this part:</p>

<pre>
> library(DESeq2)
> library(gplots)
</pre>
<p>Next we will list the files with counts into an object called sampleFiles and create an object with the condition/genotype. It is important that the order of the genotypes is the same as the order of the listed files:</p>

<pre>
> sampleFiles<-list.files(pattern="count.txt$")
#Use the same order from the sampleFiles object:
> sampleCondition <- c("ompR_KO","ompR_KO","wt","wt")
</pre>

<p>We can then create a data frame, which is a type of R object with a table structure called sampleTable:</p>

<pre>
> sampleTable <- data.frame(
  sampleName = gsub("_count.txt","",sampleFiles),
  fileName = sampleFiles,
  condition = sampleCondition
)
</pre>


<p>Notice that you defined the columns in this data frame and for sample names you run the object containing the file list through a substitution command to remove the suffix of the file names.</p>

<p>You can check the structure of the data frame just by typing:</p>
<pre>
> sampleTable
</pre>
<p>Please check if everything as we will next move to the analytical stage. You can import the HTSeq counts into a DESeqDataSet object and look at the contents of this object by running the following:</p>
<pre>
> dds <- DESeqDataSetFromHTSeqCount(
  sampleTable = sampleTable,
  design = ~ condition
)

> dds 
</pre>
<p>Pay attention to the design argument of the first function, you need to specify the grouping factor. This argument can accommodate multiple designs, this is the simplest design scheme in which you specified the name of a single column from your original dataframe.</p>

<p>A pre-filtering step can be carried out to reduce the number of genes with a count below 10. This step is not essential but it can help to reduce the size of the DESeqDataSet object. To do this, run the following:</p>

<pre>
> keep <- rowSums(counts(dds)) >= 10
> dds <- dds[keep,]
</pre>

<p>Next, setting the factor levels is important to define the reference level (control group). If this is not specified the levels will be taken in alphabetical order and the reference level will be assumed to be the first one. Below there are two options to set the levels:</p>
<pre>
> dds$condition <- factor(dds$condition, levels = c("wt","ompR_KO"))
# Notice that wt comes first as this is the natural reference group (control)
# or
> dds$condition <- relevel(dds$condition,ref="wt")
</pre>

<p>The differential expression analysis is carried out using the DESeq function that takes the DESeqDataSet object as input, does the normalization of the counts, estimates dispersion and additional statistical tests. After, the result table can be generated using the results function results:</p>

<pre>
> dds <- DESeq(dds)
> res <- results(dds, contrast = c("condition","ompR_KO","wt"))
</pre>


<p>In the second command we have just used the contrast argument to the results function which might not be necessary but enables you to specifically control the comparison being made across groups. To look at the results table just type:</p>


<pre>
> res
</pre>


<p>What do you see? Something like this:</p>

<img src="images/res_Kp.png" alt="blank space", style="max-width: 70%;">

<p>You can summarize the results by running:</p>
<pre>
> summary(res)
</pre>

<img src="images/summary_resKp.png" alt="blank space", style="max-width: 70%;">

<p>You should see the standard adjusted p-value threshold used for the summary and the number of genes with a Log2 Fold Change (LFC) above and below 0. </p>


<p>Next let’s remove results with no adjusted p-values and sort the table by this value:</p>


<pre>
> res <- res[!is.na(res$padj),]
> resOrdered <- res[order(res$pvalue),]
> resOrdered
</pre>

<p>You can construct a heatmap for the top 50 genes using the following commands:</p>

<pre>
> counts_heatmap <- counts(dds, normalized = TRUE)
> idx <- rownames(resOrdered)[1:100]
> counts_heatmap <- counts_heatmap[rownames(counts_heatmap)%in%idx,]

> counts_heatmap

> colnames(counts_heatmap) <- c("ompR_KO_2","ompR_KO_2","wt_1","wt_2")
>heatmap.2(as.matrix(counts_heatmap), scale="row", col=greenred(75), Rowv=NA, dendrogram = "col", trace="none", density.info = "none")
</pre>

<p>Another option is to visualize Gene Plot Counts for multiple genes:</p>
<pre>
> par(mfrow=c(2,3))
> plotCounts(dds,gene="rpsJ", intgroup="condition")
> plotCounts(dds,gene="VK055_3129", intgroup="condition")
> plotCounts(dds,gene="fyuA", intgroup="condition")
> plotCounts(dds,gene="cobO", intgroup="condition")
> plotCounts(dds,gene="VK055_2318", intgroup="condition")
> plotCounts(dds,gene="VK055_2733", intgroup="condition")
> par(mfrow=c(1,1))
</pre>

<p>You can also do a Principal Component Analysis of the results:</p>
<pre>
> vsdata<-vst(dds,blind=FALSE)
> z <- plotPCA(vsdata,intgroup="condition")
> z + coord_fixed(ylim=c(-10,10), xlim=c(-15,15))
</pre>

<p>Do the global transcriptomic signatures from different conditions cluster in this analysis?</p>

<p>For a Volcano Plot, you can construct a basic one:</p>

<pre>
> with(res, plot(log2FoldChange, -log10(pvalue), pch=20,main="Volcano plot",xlim=c(-8,8)))
</pre>
<p>or highlight in blue if the adjusted p-value is below 0.01 or red if the adjusted p-value is below 0.01 and the Log2 Fold Change is higher than 1:</p>
<pre>
> with(subset(res,padj<.01),points(log2FoldChange, -log10(pvalue),pch=20,col="blue"))
> with(subset(res,padj<.01 & abs(log2FoldChange)>2),points(log2FoldChange, -log10(pvalue),pch=20,col="red")) 
</pre>

<p>Moreover, you can write you results table to a csv file, or create individual files of upregulated or downregulated genes based on specific thresholds that you can decide on. For example:</p>
<pre>
> write.table(resOrdered,file="resOrdered.csv", sep=";",row.names=TRUE, quote = FALSE, col.names=TRUE)

##The following produce different files based on criteria adjusted p-value < 0.05 and Log2 Fold Change > 0.5, sorted by the adjusted p-value:
> res_sig<-subset(res,padj<0.05)
> res_sig_up<-subset(res_sig, log2FoldChange > 0.5)
> res_sig_upOrdered<-res_sig_up[order(res_sig_up$padj),]
> write.table(res_sig_upOrdered, file="res_sig_upOrdered.csv", sep=";",row.names=TRUE, quote = FALSE, col.names=TRUE)


> res_sig_down<-subset(res_sig, log2FoldChange < 0.5)
> res_sig_downOrdered<-res_sig_down[order(res_sig_down$padj),]
> write.table(res_sig_downOrdered, file="res_sig_downOrdered.csv", sep=";",row.names=TRUE, quote = FALSE, col.names=TRUE)
</pre>

<p>You can then open the CSV files in MS Excel or LibreOffice Calc and examine the results. Please note that you must adjust the column headers.</p>

<br>
<br>
<h2>References</h2>

<p>1	Wyres KL, Lam MMC, Holt KE. Population genomics of Klebsiella pneumoniae. Nat Rev Microbiol. 2020; 18: 344-359.</p>
<p>2	Spadar A, Perdigao J, Campino S, Clark TG. Genomic analysis of hypervirulent Klebsiella pneumoniae reveals potential genetic markers for differentiation from classical strains. Sci Rep. 2022; 12: 13671.</p>
<p>3	Siu LK, Yeh KM, Lin JC, Fung CP, Chang FY. Klebsiella pneumoniae liver abscess: a new invasive syndrome. Lancet Infect Dis. 2012; 12: 881-887.</p>
<p>4	Walker KA, Miller VL. The intersection of capsule gene expression, hypermucoviscosity and hypervirulence in Klebsiella pneumoniae. Curr Opin Microbiol. 2020; 54: 95-102.</p>
<p>5	Nadasy KA, Domiati-Saad R, Tribble MA. Invasive Klebsiella pneumoniae syndrome in North America. Clin Infect Dis. 2007; 45: e25-28.</p>
<p>6	Wang L, Huang X, Jin Q, et al. Two-Component Response Regulator OmpR Regulates Mucoviscosity through Energy Metabolism in Klebsiella pneumoniae. Microbiol Spectr. 2023; 11: e0054423.</p>
<p>7	Anders S, Pyl PT, Huber W. HTSeq--a Python framework to work with high-throughput sequencing data. Bioinformatics. 2015; 31: 166-169.</p>
<p>8	Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014; 15: 550.<//p>
<p>9	Robinson MD, McCarthy DJ, Smyth GK. edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics. 2010; 26: 139-140.</p>






<a href="index.html" target="_blank"><<-HOME</a>
</body>
</html>